src/Microsoft.OData.Core/Json/JsonLightInstanceAnnotationWriter.cs

```diff
@@ -84,27 +84,98 @@ internal JsonLightInstanceAnnotationWriter(ODataJsonLightValueSerializer valueSe
+                    this.WriteAndTrackInstanceAnnotation(annotation, tracker, instanceAnnotationNames, ignoreFilter, propertyName);
                 }
+            }
+            else
+            {
+                foreach (ODataInstanceAnnotation annotation in instanceAnnotations)
+                {
+                    this.WriteAndTrackInstanceAnnotation(annotation, tracker, instanceAnnotationNames, ignoreFilter, propertyName);
+                }
+            }
+        }
 
-                if (!tracker.IsAnnotationWritten(annotation.Name)
-                    && (!ODataAnnotationNames.IsODataAnnotationName(annotation.Name) || ODataAnnotationNames.IsUnknownODataAnnotationName(annotation.Name)))
+        /// <summary>
+        /// Write instance annotation if not already written.
+        /// </summary>
+        /// <param name="annotation">The instance annotation to write.</param>
+        /// <param name="tracker">The tracker to track if instance annotations are written.</param>
+        /// <param name="instanceAnnotationNames">Set used to detect a duplicate annotation.</param>
+        /// <param name="ignoreFilter">Whether to ignore the filter in settings.</param>
+        /// <param name="propertyName">The name of the property this instance annotation applies to.</param>
+        private void WriteAndTrackInstanceAnnotation(
+            ODataInstanceAnnotation annotation,
+            InstanceAnnotationWriteTracker tracker,
+            HashSet<string> instanceAnnotationNames,
+            bool ignoreFilter = false,
+            string propertyName = null)
+        {
+
+            if (!instanceAnnotationNames.Add(annotation.Name))
+            {
+                throw new ODataException(ODataErrorStrings.JsonLightInstanceAnnotationWriter_DuplicateAnnotationNameInCollection(annotation.Name));
+            }
+
+            if (!tracker.IsAnnotationWritten(annotation.Name)
+                        && (!ODataAnnotationNames.IsODataAnnotationName(annotation.Name) || ODataAnnotationNames.IsUnknownODataAnnotationName(annotation.Name)))
+            {
+                this.WriteInstanceAnnotation(annotation, ignoreFilter, propertyName);
+                tracker.MarkAnnotationWritten(annotation.Name);
+            }
+        }
+
+        /// <summary>
+        /// Write all the instance annotations specified in <paramref name="instanceAnnotations"/> for the
+        /// undecalared property called <paramref name="propertyName"/>
+        /// </summary>
+        /// <param name="instanceAnnotations">The collection of instance annotations</param>
+        /// <param name="propertyName">The name of the property the instance annotations apply to</param>
+        private void WriteInstanceAnnotationsForUndeclaredProperty(ICollection<ODataInstanceAnnotation> instanceAnnotations, string propertyName)

```

Comment:
The convention across the repo has been to have `private` methods appear after `public` and `internal`

Reply:
Would it have been better to have the caller pass the value for `ignoreFilter` instead of hardcoding it to `true` in this private method? Though it's private so that would be easy to change...